src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestState.cs

```diff
@@ -55,6 +55,24 @@ public object Lock
             }
         }
 
+        public void ClearSendRequestState()
+        {
+            // Since WinHttpRequestState has a self-referenced strong GCHandle, we
```

Comment:
I thought about having two state objects, one for the request/response phase and another for the responsestream reading phase. But then we'd have to make WinHTTP calls to update the context object and make sure that the logic is atomic for rolling-back on errors.

I am considering restructuring this in some way but differing that until I do #2501.